src/status_im/data_store/chats.cljs

```diff
@@ -31,121 +37,124 @@
 
 (defn marshal-membership-updates [updates]
   (mapcat (fn [{:keys [signature events from]}]
-            (map #(assoc %
-                         :id (event-id %)
-                         :signature signature
-                         :from from) events)) updates))
+            (map #(-> %
+                      (assoc
+                       :clockValue (:clock-value %)
+                       :id (event-id %)
+                       :signature signature
+                       :from from)
+                      (dissoc :clock-value)) events)) updates))
 
 (defn unmarshal-membership-updates [chat-id updates]
   (->> updates
-       vals
        (group-by :signature)
        (map (fn [[signature events]]
-              {:events (map #(-> (dissoc % :signature :from :id)
+              {:events (map #(-> %
+                                 (assoc :clock-value (:clockValue %))
+                                 (dissoc :signature :from :id :clockValue)
                                  remove-empty-vals) events)
                :from  (-> events first :from)
                :signature signature
                :chat-id chat-id}))))
 
-(defn- normalize-chat [{:keys [chat-id] :as chat}]
+(defn type->rpc [{:keys [public? group-chat] :as chat}]
+  (assoc chat :chatType (cond
+                          public? public-chat-type
+                          group-chat private-group-chat-type
+                          :else one-to-one-chat-type)))
+
+(defn rpc->type [{:keys [chatType] :as chat}]
+  (cond
+    (= public-chat-type chatType) (assoc chat :public? true :group-chat true)
+    (= private-group-chat-type chatType) (assoc chat :public? false :group-chat true)
+    :else (assoc chat :public? false :group-chat false)))
+
+(defn- marshal-members [{:keys [admins contacts members-joined chatType] :as chat}]
+  (cond-> chat
+    (= chatType private-group-chat-type)
+    (assoc :members (map #(hash-map :id %
+                                    :admin (boolean (admins %))
+                                    :joined (boolean (members-joined %))) contacts))
+    :always
+    (dissoc :admins :contacts :members-joined)))
+
+(defn- unmarshal-members [{:keys [members chatType] :as chat}]
```

Comment:
why do we need all these data transformations on react side? `->rpc` and `<-rpc` ?

Reply:
This data structures come from `status-protocol-go` which one of its goals is to be agnostic to status-react,  so there data-structures are not quite the same, so they need to be "massaged" in shape, we can do this either in status-react or status-go, for now I picked status-react as my understanding is that `status-go` as well should not have much knowledge of status-react (i.e it should for example serve `camelCase` json rather than `kebab-case` which is very clojure specific), but I am happy with both, in any case if we want to change I'd rather do it separately, it should be pretty straightforward to move to status-go.